examples/nodes/hf_datsets.py

```diff
@@ -0,0 +1,82 @@
+from datasets import Dataset  # make sure that you have huggingface datasets installed
+
+data = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
+
+from torchdata.nodes.adapters import IterableWrapper
+from torchdata.nodes.batch import Batcher
+from torchdata.nodes.map import Mapper, ParallelMapper
+from torchdata.nodes.prefetch import Prefetcher
+
+ds = Dataset.from_dict({"data": data})
+
+source_node = IterableWrapper(ds)
+
+print("==" * 20)
```

Comment:
One thing I had in mind with /examples, we don't expect users to really run the code, but mostly use it as a reference implementation to port over their workflows. I wonder if we can write this file such that users can see all the relevant nodes linked together clearly?

Reply:
Yes, sure.
I will port these examples to a Jupyter notebook with print outputs in the file so it is easier to get the outputs.
Will also add some interleaved examples where we use multiple node functionalities together.